src/HomeAssistant.cpp

```diff
@@ -67,45 +68,61 @@ FuFarmHomeAssistant::FuFarmHomeAssistant(Client &client) :
   // Setup sensors with the relevant information
   // Device classes are listed at https://www.home-assistant.io/integrations/sensor/#device-class
 #ifdef HAVE_WATER_LEVEL_STATE
-  // TODO: find a better device class for this
   waterLevel.setDeviceClass("moisture");
+  waterLevel.setIcon("mdi:cup-water");
+  waterLevel.setName("Water Tank Level");
   waterLevel.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_LIGHT
+  // TODO: the sensor we are using does not support lux, we need to consider using a custom class
   light.setDeviceClass("illuminance");
+  light.setUnitOfMeasurement("lx");
   light.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #if defined(HAVE_DHT22) || defined(HAVE_AHT20)
   temperature.setDeviceClass("temperature");
+  temperature.setUnitOfMeasurement("°C");
   temperature.setExpireAfter(EXPIRE_AFTER_SECONDS);
 
   humidity.setDeviceClass("humidity");
+  humidity.setUnitOfMeasurement("%");
   humidity.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_ENS160
   aqi.setDeviceClass("aqi");
   aqi.setExpireAfter(EXPIRE_AFTER_SECONDS);
+  // no need to set unit for AQI, it is documented as unitless
 
   tvoc.setDeviceClass("volatile_organic_compounds_parts");
+  tvoc.setUnitOfMeasurement("ppb");
   tvoc.setExpireAfter(EXPIRE_AFTER_SECONDS);
 
   eco2.setDeviceClass("carbon_dioxide");
+  eco2.setName("Carbon Dioxide (Equivalent)");
+  eco2.setUnitOfMeasurement("ppm");
   eco2.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_FLOW
   flow.setDeviceClass("volume_flow_rate");
+  flow.setUnitOfMeasurement("L/min");
   flow.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_TEMP_WET
   liquidtemp.setDeviceClass("temperature");
+  liquidtemp.setName("Liquid Temperature");
+  liquidtemp.setUnitOfMeasurement("°C");
   liquidtemp.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_CO2
   co2.setDeviceClass("carbon_dioxide");
+  co2.setUnitOfMeasurement("ppm");
   co2.setExpireAfter(EXPIRE_AFTER_SECONDS);
 #endif
 #ifdef HAVE_EC
-  ec.setDeviceClass("ec");
+  // As of 2025-Mar-28, Home Assistant does not have a device class for EC, we create a custom one
```

Comment:
It's on my list is to add EC as a device class to home assistant. I've not got around to doing it as from the bits I've looked at, it'd be fairly involved to set up a HA development environment and be confident of submitting it properly. There's also a bug with the volume_flow_rate  implementation (the icon just shows up as a generic one with no units), so my plan was to look into that as a way of understanding how to implement EC.

Reply:
I hadn't thought of submitting a PR to Home Assistant, you are likely better placed to do it then we can update the code once done. In the meantime, I have added the icon for `volume_flow_rate`.